### PR TITLE
add: Night Of The Dead 

### LINKED ISSUE #36: [Egg Request]: Night Of The Dead
### Does this expand an already existing service

No

### Link to a game or software

https://store.steampowered.com/app/1377380/Night_of_the_Dead/

### Links for the download

https://store.steampowered.com/app/1377380/Night_of_the_Dead/

### Links for the install docs

https://store.steampowered.com/news/app/1377380/view/2799507523521313093

### PR DESCRIPTION
**Pull request by QuintenQVD0**
# Description

closes #36 

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [ ] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [ ] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [ ] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

### CODE REVIEW

**Comment by scootz** [comment]
I tested the new egg, did it twice to make sure.  I get a 404 error where it tries to curl the gamesettings.ini and this is the resulting file below. I've deleted the obvious settings but other than that I left it as it was. 

image

File: night_of_the_dead/egg-pterodactyl-night-of-the-dead.json [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|string|max:32",
+            "field_type": "text"
+        },
+        {
+            "name": "Winetricks Run",
+            "description": "",
+            "env_variable": "WINETRICKS_RUN",
+            "default_value": "vcrun2019 corefonts",
```
Comment (gOOvER) [inline night_of_the_dead/egg-pterodactyl-night-of-the-dead.json:147]: use vcrun2022. it cointains all other versions

**Review by gOOvER — CHANGES_REQUESTED** [review]

**Comment by gOOvER** [comment]
> I tested the new egg, did it twice to make sure. I get a 404 error where it tries to curl the serversettings.ini and this is the resulting file below. I've deleted the obvious settings but other than that I left it as it was.
> 
> image

yes, because its not in the repo yet

**Comment by QuintenQVD0** [comment]
> I tested the new egg, did it twice to make sure. I get a 404 error where it tries to curl the serversettings.ini and this is the resulting file below. I've deleted the obvious settings but other than that I left it as it was.
> 
> image

Yes that is because the file will only exits after the egg has been merged, thats normal

File: night_of_the_dead/egg-pterodactyl-night-of-the-dead.json [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+            "default_value": "Normal",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|string|in:Normal,Peaceful,Easy,Hard,Survival,Challenge,Legend",
+            "field_type": "text"
+        },
+        {
+            "name": "Enable pvp",
+            "description": "1=PVE 0=PVP",
+            "env_variable": "ENABLE_PVP",
```
Comment (scootz) [inline night_of_the_dead/egg-pterodactyl-night-of-the-dead.json:126]: ENABLE_PVP should be changed to PVP to match what is in the configuration files section.

File: night_of_the_dead/egg-pterodactyl-night-of-the-dead.json [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            "default_value": "1",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|boolean",
+            "field_type": "text"
+        },
+        {
+            "name": "Savegame Name",
+            "description": "",
+            "env_variable": "SAVEGAME_NAME",
```
Comment (scootz) [inline night_of_the_dead/egg-pterodactyl-night-of-the-dead.json:136]: This SAVEGAME_NAME does not match what is in the configuration files section which has SERVER_SAVEGAME.

File: night_of_the_dead/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Nigh of the dead
```
Comment (scootz) [inline night_of_the_dead/README.md:1]: Typo,  Night.
```suggestion
# Night of the dead

```

File: night_of_the_dead/ServerSettings.ini [hunk 5]
```diff
@@ -0,0 +1,94 @@
+[ServerSettings]
+ServerName=
+Password=
+MaxPlayers=16
+[SystemSettings]
+IngameAdminPassword=
+DefaultMessageOfTheDay=
+; INSTALL_DIR/Saved/SaveGames/SAVE_NAME
+SaveName=/home/container/Saved/SaveGames/
```
Comment (scootz) [inline night_of_the_dead/ServerSettings.ini:9]:
```suggestion
SaveName=/home/container/LF/Saved/SaveGames/

```

File: night_of_the_dead/egg-night-of-the-dead.json [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+        "steam_disk_space"
+    ],
+    "docker_images": {
+        "wine_latest": "ghcr.io\/parkervcp\/yolks:wine_latest"
+    },
+    "file_denylist": [],
+    "startup": "xvfb-run --auto-servernum wine \/home\/container\/LF\/Binaries\/Win64\/LFServer-Win64-Shipping.exe ?listen -Port={{SERVER_PORT}} -QueryPort={{QUERY_PORT}} -console -log --CRASHREPORTS {{OTHER_ARGS}}",
+    "config": {
+        "files": "{\n    \".\/LF\/Saved\/Config\/ServerSettings.ini\": {\n        \"parser\": \"ini\",\n        \"find\": {\n            \"ServerSettings.ServerName\": \"{{server.environment.SERVER_NAME}}\",\n            \"ServerSettings.Password\": \"{{server.environment.SERVER_PASSWORD}}\",\n            \"SystemSettings.SaveName\": \"\/home\/container\/Saved\/SaveGames\/{{server.environment.SERVER_SAVEGAME}}\",\n            \"SystemSettings.IngameAdminPassword\": \"{{server.environment.SERVER_ADMIN_PASSWORD}}\",\n            \"SystemSettings.DefaultMessageOfTheDay\": \"{{server.environment.SERVER_MOTD}}\",\n            \"GameSettings.Difficulty\": \"{{server.environment.DIFFICULTY}}\",\n            \"GameSettings\/General.MODSelect\": \"{{server.environment.PVP}}\"\n        }\n    }\n}",
```
Comment (scootz) [inline night_of_the_dead/egg-night-of-the-dead.json:21]: SystemSettings.SaveName should have a path change  including the 'LF' directory in the path. /home/container/LF/Saved/SaveGames/{{server.environment.SERVER_SAVEGAME}}

File: night_of_the_dead/egg-night-of-the-dead.json [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
...EAM_PASS} ${STEAM_AUTH} $( [[ \"${WINDOWS_INSTALL}\" == \"1\" ]] && printf %s '+@sSteamCmdForcePlatformType windows' ) +app_update ${SRCDS_APPID} $( [[ -z ${SRCDS_BETAID} ]] || printf %s \"-beta ${SRCDS_BETAID}\" ) $( [[ -z ${SRCDS_BETAPASS} ]] || printf %s \"-betapassword ${SRCDS_BETAPASS}\" ) ${INSTALL_FLAGS} validate +quit ## other flags may be needed depending on install. looking at you cs 1.6\r\n\r\n## set up 32 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk32\r\ncp -v linux32\/steamclient.so ..\/.steam\/sdk32\/steamclient.so\r\n\r\n## set up 64 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk64\r\ncp -v linux64\/steamclient.so ..\/.steam\/sdk64\/steamclient.so\r\n\r\n## add below your custom commands if needed\r\n## make needed dir\r\nmkdir -p \/mnt\/server\/Saved\/SaveGames\r\nmkdir -p \/mnt\/server\/LF\/Saved\/Config\r\n\r\ncurl -sSL -o \/mnt\/server\/LF\/Saved\/Config\/ServerSettings.ini https:\/\/raw.githubusercontent.com\/pelican-eggs\/games-steamcmd\/main\/night_of_the_dead\/ServerSettings.ini\r\n\r\n## install end\r\necho \"-----------------------------------------\"\r\necho \"Installation completed...\"\r\necho \"-----------------------------------------\"",
```
Comment (scootz) [inline night_of_the_dead/egg-night-of-the-dead.json:28]: Where it makes the savegames path, should include 'LF' directory.   mkdir -p /mnt/server/LF/SaveGames

File: night_of_the_dead/egg-night-of-the-dead.json [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|string|in:Normal,Peaceful,Easy,Hard,Survival,Challenge,Legend",
+            "field_type": "text"
+        },
+        {
+            "sort": 10,
+            "name": "Enable pvp",
+            "description": "1=PVE 0=PVP",
+            "env_variable": "ENABLE_PVP",
```
Comment (scootz) [inline night_of_the_dead/egg-night-of-the-dead.json:137]: This should be changed to 'PVP' to match configuration files section variable usage.

File: night_of_the_dead/egg-night-of-the-dead.json [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|boolean",
+            "field_type": "text"
+        },
+        {
+            "sort": 11,
+            "name": "Savegame Name",
+            "description": "",
+            "env_variable": "SAVEGAME_NAME",
```
Comment (scootz) [inline night_of_the_dead/egg-night-of-the-dead.json:148]: make this match configuration files section variable usage.
```suggestion
            "env_variable": "SERVER_SAVEGAME",
```

File: night_of_the_dead/egg-pterodactyl-night-of-the-dead.json [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+    "features": [
+        "steam_disk_space"
+    ],
+    "docker_images": {
+        "wine_latest": "ghcr.io\/parkervcp\/yolks:wine_latest"
+    },
+    "file_denylist": [],
+    "startup": "xvfb-run --auto-servernum wine \/home\/container\/LF\/Binaries\/Win64\/LFServer-Win64-Shipping.exe ?listen -Port={{SERVER_PORT}} -QueryPort={{QUERY_PORT}} -console -log --CRASHREPORTS {{OTHER_ARGS}}",
+    "config": {
+        "files": "{\n    \".\/LF\/Saved\/Config\/ServerSettings.ini\": {\n        \"parser\": \"ini\",\n        \"find\": {\n            \"ServerSettings.ServerName\": \"{{env.SERVER_NAME}}\",\n            \"ServerSettings.Password\": \"{{env.SERVER_PASSWORD}}\",\n            \"SystemSettings.SaveName\": \"\/home\/container\/Saved\/SaveGames\/{{env.SERVER_SAVEGAME}}\",\n            \"SystemSettings.IngameAdminPassword\": \"{{env.SERVER_ADMIN_PASSWORD}}\",\n            \"SystemSettings.DefaultMessageOfTheDay\": \"{{env.SERVER_MOTD}}\",\n            \"GameSettings.Difficulty\": \"{{env.DIFFICULTY}}\",\n            \"GameSettings\/General.MODSelect\": \"{{env.PVP}}\"\n        }\n    }\n}",
```
Comment (scootz) [inline night_of_the_dead/egg-pterodactyl-night-of-the-dead.json:20]:
```suggestion
        "files": "{\n    \".\/LF\/Saved\/Config\/ServerSettings.ini\": {\n        \"parser\": \"ini\",\n        \"find\": {\n            \"ServerSettings.ServerName\": \"{{env.SERVER_NAME}}\",\n            \"ServerSettings.Password\": \"{{env.SERVER_PASSWORD}}\",\n            \"SystemSettings.SaveName\": \"\/home\/container\/LF\/Saved\/SaveGames\/{{env.SERVER_SAVEGAME}}\",\n            \"SystemSettings.IngameAdminPassword\": \"{{env.SERVER_ADMIN_PASSWORD}}\",\n            \"SystemSettings.DefaultMessageOfTheDay\": \"{{env.SERVER_MOTD}}\",\n            \"GameSettings.Difficulty\": \"{{env.DIFFICULTY}}\",\n            \"GameSettings\/General.MODSelect\": \"{{env.PVP}}\"\n        }\n    }\n}",
```

File: night_of_the_dead/egg-pterodactyl-night-of-the-dead.json [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
...EAM_PASS} ${STEAM_AUTH} $( [[ \"${WINDOWS_INSTALL}\" == \"1\" ]] && printf %s '+@sSteamCmdForcePlatformType windows' ) +app_update ${SRCDS_APPID} $( [[ -z ${SRCDS_BETAID} ]] || printf %s \"-beta ${SRCDS_BETAID}\" ) $( [[ -z ${SRCDS_BETAPASS} ]] || printf %s \"-betapassword ${SRCDS_BETAPASS}\" ) ${INSTALL_FLAGS} validate +quit ## other flags may be needed depending on install. looking at you cs 1.6\r\n\r\n## set up 32 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk32\r\ncp -v linux32\/steamclient.so ..\/.steam\/sdk32\/steamclient.so\r\n\r\n## set up 64 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk64\r\ncp -v linux64\/steamclient.so ..\/.steam\/sdk64\/steamclient.so\r\n\r\n## add below your custom commands if needed\r\n## make needed dir\r\nmkdir -p \/mnt\/server\/Saved\/SaveGames\r\nmkdir -p \/mnt\/server\/LF\/Saved\/Config\r\n\r\ncurl -sSL -o \/mnt\/server\/LF\/Saved\/Config\/ServerSettings.ini https:\/\/raw.githubusercontent.com\/pelican-eggs\/games-steamcmd\/main\/night_of_the_dead\/ServerSettings.ini\r\n\r\n## install end\r\necho \"-----------------------------------------\"\r\necho \"Installation completed...\"\r\necho \"-----------------------------------------\"",
```
Comment (scootz) [inline night_of_the_dead/egg-pterodactyl-night-of-the-dead.json:27]: Again, in here, change where it makes the savegame directory to include /LF/ in the path.    mkdir -p \/mnt\/server\/LF\/Saved\/SaveGames

**Comment by scootz** [comment]
There is also this that happens with repeated paths i'm not sure how to fix this.
image

**Comment by QuintenQVD0** [comment]
@scootz I made the request changes and I also made the console output work, please re-test with a new server (not updating an existing one) the savegame path should be fixt

**Comment by scootz** [comment]
Works good.  There is still the issue with the saved game path.
image

**Comment by QuintenQVD0** [comment]
> Works good. There is still the issue with the saved game path. The panel still thinks its crashing when stopping the server.
> 
> image

that thinking that is crashed is something I also have (on pelican no idea if you are on that or ptero).

the savegames I did not have one as I do not own the game so I could not join ofc.

**Comment by scootz** [comment]
The game works, its just confusing when anyone would want to restore a saved game.  Might I suggest changing in the configuration files section the ServerSettings.SaveName to omit the /home/container/LF/Saved/SaveGames/  and just have the variable {{server.environment.SERVER_SAVEGAME}}

This is running on pelican.

**Comment by QuintenQVD0** [comment]
> The game works, its just confusing when anyone would want to restore a saved game. Might I suggest changing in the configuration files section the ServerSettings.SaveName to omit the /home/container/LF/Saved/SaveGames/ and just have the variable {{server.environment.SERVER_SAVEGAME}}
> 
> This is running on pelican.

you can try that if you want, It sees like you know where to find it.

**Comment by scootz** [comment]
Ok yea just having the variable works.  So the only thing left I think is pelican thinking its crashed when stopping the server.

**Comment by QuintenQVD0** [comment]
> Ok yea just having the variable works. So the only thing left I think is pelican thinking its crashed when stopping the server.

that's an issue at panel level not egg as I read that more than 1 egg have this.

**Comment by QuintenQVD0** [comment]
> Ok yea just having the variable works. So the only thing left I think is pelican thinking its crashed when stopping the server.

Change made, should be good to go now!

**Comment by gOOvER** [comment]
a long discussion for a really simple egg :D I have an egg since release. Its really easy to create

**Comment by gOOvER** [comment]
the savegame fix was not needed. Users are able to rename it before uploading it. An egg is only for 1 server and not for a lot of savegames ;)

**Comment by QuintenQVD0** [comment]
> able to rename it before u

it had a double path in it because savegame name was set to a path not a filename and auto prefixt by the parser.
It is fine this way now.

**Review by scootz — APPROVED** [review]
Everything so far looks good to me and works fine in the panel.  Game works as expected. Ty!

**Comment by WSG-ViViV** [comment]
Hello All, I have tried this egg. Need it something special ?? I have this error :
[2024.06.16-15.56.10:655][604]LogStreaming: Error: Found 1 dependent packages...
[2024.06.16-15.56.10:666][604]LogStreaming: Error:   /Game/maps/LFMap/LFMap_Land_140
And noway to start it. 
Everything is up to date and it's not my first install ^^

**Comment by gOOvER** [comment]
> Hello All, I have tried this egg. Need it something special ?? I have this error : [2024.06.16-15.56.10:655][604]LogStreaming: Error: Found 1 dependent packages... [2024.06.16-15.56.10:666][604]LogStreaming: Error: /Game/maps/LFMap/LFMap_Land_140 And noway to start it. Everything is up to date... Let me know if I can do something for help

this error is normally on a "notice"

**Comment by WSG-ViViV** [comment]
It's the last line in logs before crash.
[LF.log](https://github.com/user-attachments/files/15858972/LF.log)

**Comment by gOOvER** [comment]
problem with your home network

`LogStreaming: Warning: Failed to read file 'Z:/home/container/Cloud/IoStoreOnDemand.ini' error.`

**Comment by WSG-ViViV** [comment]
Ook, that don't help me --' 
I will try to know why it don't work with me.
I have another server on same node and it work well...

**Comment by gOOvER** [comment]
> Ook, that don't help me --' I will try to know why it don't work with me. I have another server on same node and it work well...

could be a failed installation

**Comment by WSG-ViViV** [comment]
Ook i have try on another node and it work well
Thank's for help =))

**Pull request merged**